docs/imviz/plugins.rst

```diff
@@ -181,15 +181,18 @@ an interactively selected region. A typical workflow is as follows:
    object of interest using its center of mass, if you wish.
    Depending on the object, it may take several iterations for re-centering
    to converge, or it may never converge at all.
+
+   .. note::
+
+       You cannot use annulus region as aperture (an exception will be thrown)
+       but you may use it for background (see below).
+
```

Comment:
this error might be confusing.  Can we either exclude the annulus subsets from the dropdown, or check immediately on selection and raise a validation error instead?  Or at the least, can we provide a more end-user-friendly message that states that annulus regions are not valid for the aperture?
<img width="1031" alt="Screen Shot 2023-06-30 at 12 45 06 PM" src="https://github.com/spacetelescope/jdaviz/assets/877591/c6ae4208-c100-42ca-9f1a-7b25a11fbc36">

Reply:
@kecnry , since you brought it up, is there a way to add custom filter here without having to refactor that class itself? I thought about having the aperture dropdown to skip incompatible shape and not show them but not sure how to do it.

https://github.com/spacetelescope/jdaviz/blob/b6feaa6fd3fedf7e2f7acedee7e03a67f9f906d6/jdaviz/core/template_mixin.py#L902

I vaguely remember you implemented some sort of filtering somewhere...

Reply:
this is more involved than I had hoped.  Feel free to leave as-is for now and open a follow-up ticket so we can generalize filtering to the subset component.

Reply:
Okay I opened https://github.com/spacetelescope/jdaviz/issues/2280